File: solutions_5636311922769920_0/C++/mcprobie7/QD.cpp
#include <stdio.h>
#include  <iostream>

using namespace std;

int main()
{
    freopen("D-small-attempt0.in", "r", stdin);
    freopen("outputD.txt", "w", stdout);

    int nT;

    cin >> nT;

    for (int iT = 1; iT <= nT; iT++)
    {
        int K, C, S;

        cin >> K >> C >> S;

        long long K0 = 1;
        for (int i = 0; i < C - 1; i++) K0 = K0 * K;

        cout << "Case #" << iT << ":";

        long long K1 = 1;
        for (int i = 0; i < S; i++)
        {
            cout << " " << K1;
            K1 += K0;

        }
        cout << endl;


    }
    return 0;
}

File: solutions_5636311922769920_1/Python/Taizo/q_D.py

# -*- coding: cp932 -*-

import sys

#inputFile = "D-sample.in"
#inputFile = "D-small-attempt0.in"
inputFile = "D-large.in"


f = open(inputFile)
sys.stdout = open(inputFile.replace(".in", ".txt"), 'w')
tc_num = int(f.readline().rstrip())


def solve(k, c, s):
    if c * s < k:
        return " IMPOSSIBLE"

    ans = []
    atten = 1

    while True:
        cur = atten
        atten += 1
        for fn in range(2, c+1):
            #sys.stderr.write("fn " + str(fn) + " atten = " + str(atten) + " curr = " + str(cur) + "\n")
            #print("fn " + str(fn) + " atten = " + str(atten) + " curr = " + str(cur))
            if atten > k:
                cur = (cur - 1) * k + 1
            else:
                cur = (cur - 1) * k + atten
                atten += 1

        ans.append(cur)
        if atten > k:
            break

    ansStr = ""
    for ansNum in ans:
        ansStr += " " + str(ansNum)

    return ansStr


for tc in range(tc_num):
    l = f.readline().split()
    K = int(l[0])
    C = int(l[1])
    S = int(l[2])

    print("Case #" + str(tc + 1) + ":" + solve(K, C, S))

